pkg/compactor/blocks_cleaner.go

```diff
@@ -628,12 +637,20 @@ func (c *BlocksCleaner) cleanUser(ctx context.Context, userLogger log.Logger, us
 		level.Info(userLogger).Log("msg", "finish cleaning partial blocks", "duration", time.Since(begin), "duration_ms", time.Since(begin).Milliseconds())
 	}
 
-	// Upload the updated index to the storage.
-	begin = time.Now()
-	if err := bucketindex.WriteIndex(ctx, c.bucketClient, userID, c.cfgProvider, idx); err != nil {
-		return err
+	if idx.IsEmpty() && len(partials) == 0 {
+		level.Info(userLogger).Log("msg", "deleting bucket index since it is empty")
+		if err := bucketindex.DeleteIndex(ctx, c.bucketClient, userID, c.cfgProvider); err != nil {
```

Comment:
Unrelated to this PR. Do we need to handle `AccessDenied`error in `bucketindex.DeleteIndex` and `bucketindex.DeleteIndexSyncStatus`?

I see they only handle object not found

Reply:
Cortex itself cannot handle `AccessDenied`. Once it happens, cleaner should fail the process and enduser should take some action on it.